pgdrive/envs/pgdrive_env.py

```diff
@@ -634,6 +638,13 @@ def saver(self, v_id: str, vehicle: BaseVehicle, actions):
         vehicle.step_info.update(saver_info)
         return steering, throttle
 
+    def _sync_reward_config_in_single_agent_env(self):
+        set_1 = set(self.config.keys())
+        set_2 = set(BaseVehicle.get_vehicle_config().keys())
+        interesct = set_1.intersection(set_2)
+        for k in list(interesct):
+            self.config["target_vehicle_configs"][DEFAULT_AGENT][k] = self.config[k]
+
```

Comment:
This is wrong! I think you need to introduce some merge_dict helper function rather than just compute the diff in keys.